### PR TITLE
Validate LabelName/LabelSet on JSON unmarshaling.

### PR DESCRIPTION
**Pull request by fabxc**

### CODE REVIEW

**Comment by brian-brazil** [comment]
:+1:

File: model/labelset.go [hunk 1]
```diff
@@ -62,3 +63,21 @@ func (l LabelSet) MergeFromMetric(m Metric) {
...
 }
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (l *LabelSet) UnmarshalJSON(b []byte) error {
+	var m map[LabelName]LabelValue
+	if err := json.Unmarshal(b, &m); err != nil {
+		return err
+	}
+	// encoding/json only unmarshals maps of the form map[string]T. It does not
+	// detect that LabelName is a string and does not call its UnmarshalJSON method.
```
Comment (beorn7) [inline model/labelset.go]: Isn't that misworded? You want to tell us that LabelName is treated as a string and not as  LabelName by encoding/json, don't you?

Comment (fabxc) [inline model/labelset.go]: Fixed.

**Comment by beorn7** [comment]
:+1:

**Pull request merged**